icu4c/source/i18n/msgfmt.cpp

```diff
@@ -1639,6 +1639,7 @@ void MessageFormat::cacheExplicitFormats(UErrorCode& status) {
         Formattable::Type formattableType;
 
         switch (argType) {
+        case UMSGPAT_ARG_TYPE_SELECT:
```

Comment:
This change seems like purely a matter of taste. The switch cases are probably in the order of their enum declaration, which is also a logical grouping. Better to leave as is.

Some of these changes, if we did want to make them, should be done in parallel in both C++ and Java, to keep the code in sync.